cmd/secretless-broker/main.go

```diff
@@ -3,32 +3,53 @@ package main
 import (
 	"flag"
 	"log"
+	"strings"
 
-	"github.com/cyberark/secretless-broker/internal/pkg/plugin"
 	yaml "gopkg.in/yaml.v1"
+
+	"github.com/cyberark/secretless-broker/internal/pkg/plugin"
+	"github.com/cyberark/secretless-broker/pkg/secretless/config"
 )
 
+func parseConfigManagerSpec(configManagerSpec string) (configManagerID string, configFilterSpec string) {
+	if len(configManagerSpec) == 0 {
+		return "", ""
+	}
+
+	configManagerSpecItems := strings.SplitN(configManagerSpec, "#", 2)
+
+	if len(configManagerSpecItems) < 1 {
+		log.Fatalf("ERROR: Manager config must be supplied in '<manager_id>[#<filter_spec>]' form")
+	}
+
+	configManagerID = configManagerSpecItems[0]
+
+	if len(configManagerSpecItems) > 1 {
+		configFilterSpec = configManagerSpecItems[1]
+	}
+
+	return
+}
+
 func main() {
 	log.Println("Secretless starting up...")
 
-	pluginDir := flag.String("p", "/usr/local/lib/secretless", "Directory containing Secretless plugins")
 	configFile := flag.String("f", "secretless.yml", "Location of the configuration file.")
+	pluginDir := flag.String("p", "/usr/local/lib/secretless", "Directory containing Secretless plugins")
+	configManagerSpec := flag.String("config-mgr", "", "Specify a configuration manager to use and filterSpec (e.g. k8s/crd#test-meta-id).")
```

Comment:
please add a comment that this is optional and the default file-based config mgr will run without